Chat socket

OlgaSavina opened the pull request:
## Describe your changes
- I did required endpoints and sockets to backend, also add opportunity to add users when user creates room
## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-209l)

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [ ] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [ ] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).


### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [ ] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

Commit:
feat: add chat sockets

Commit:
feat: add endpoints to room

Commit:
fix: update create room

Commit:
fix: formating room controller

lyudad commented:
<img width="320" alt="Screenshot 2024-01-03 at 18 22 13" src="https://github.com/ZenBit-Tech/ZenBitRock_be/assets/25103090/a1a3789a-383a-4660-bdef-39f9b51df865">

Commit:
fix: change structure

OlgaSavina commented:
> <img alt="Screenshot 2024-01-03 at 18 22 13" width="320" src="https://private-user-images.githubusercontent.com/25103090/293984612-a1a3789a-383a-4660-bdef-39f9b51df865.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQzODIxMDUsIm5iZiI6MTcwNDM4MTgwNSwicGF0aCI6Ii8yNTEwMzA5MC8yOTM5ODQ2MTItYTFhMzc4OWEtMzgzYS00NjYwLWJkZWYtMzlmOWI1MWRmODY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDE1MjMyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxZGNkMjgyYjA0ZjA1YjIwMGU4MDYzMjA0NTk0ODNjY2RiNDUzM2EwOTBjMzUwYjAyNDFiZGUyYTBiYmM2MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SfSp1nJCIWYtDzqUPJzwz4N4sDv2jpFrNPL_e_a8odM">

I have updated structure, and also renamed room table to chat. The result :
![structure](https://github.com/ZenBit-Tech/ZenBitRock_be/assets/71349433/72a459d9-70e3-46c4-a5cf-08b38400c7a1)
![chat](https://github.com/ZenBit-Tech/ZenBitRock_be/assets/71349433/34f2987f-2e25-4eee-a193-2aea10611bae)
